src/goDebugConfiguration.ts

```diff
@@ -80,7 +80,11 @@ export class GoDebugConfigurationProvider implements vscode.DebugConfigurationPr
 			debugConfiguration['dlvLoadConfig'] = dlvConfig['dlvLoadConfig'];
 		}
 
-		debugConfiguration['dlvToolPath'] = getBinPath('dlv');
+		if (!debugConfiguration['dlvToolPath']) {
+			debugConfiguration['dlvToolPath'] = getBinPath('dlv');
+		} else {
+			debugConfiguration['dlvToolPath'] = resolvePath(debugConfiguration['dlvToolPath']);

```

Comment:
Few cases to take care of
- what if user given path is not a valid absolute or relative path?
- where are we expanding user given relative path to absolute path
- check for if resolved path actually exists